buildkitd/buildkitd.go

```diff
@@ -356,3 +362,44 @@ func GetAvailableImageID(ctx context.Context, image string) (string, error) {
 	}
 	return string(output), nil
 }
+
+// CheckCompatibility runs all avaliable compatibility checks before starting the buildkitd daemon.
+func CheckCompatibility(ctx context.Context, settings Settings) error {
+	isNamespaced, err := isNamespacedDocker(ctx)
+	if isNamespaced {
+		return errors.New(`user namespaces are enabled, run with --bkd-arg "--userns=host" to disable`)
+	} else if err != nil {
+		return errors.Wrap(err, "failed compatibilty check")
+	}
+
+	isRootless, err := isRootlessDocker(ctx)
+	if isRootless {
+		return errors.New(`rootless docker detected. Compatibility is limited. Run with --bkd-arg "" to give it a shot`)
```

Comment:
The error message seems incomplete?

Reply:
Hm. The intent wasn't to make it seem empty. Specifying it is how you override the compatibility check; see my comments in my commentary review. That said, I just realized that it will sill have `len(additionalArgs)=0`, so we need something we can pass in there thats more or less a no-op :thinking: